src/main/java/org/opensearch/neuralsearch/search/HitsThresholdChecker.java

```diff
@@ -26,15 +26,15 @@ public HitsThresholdChecker(int totalHitsThreshold) {
         this.totalHitsThreshold = totalHitsThreshold;
     }
 
-    protected void incrementHitCount() {
+    public void incrementHitCount() {
         ++hitCount;
     }
 
-    protected boolean isThresholdReached() {
+    public boolean isThresholdReached() {
         return hitCount >= getTotalHitsThreshold();
     }
 
-    protected ScoreMode scoreMode() {
+    public ScoreMode scoreMode() {
         return ScoreMode.TOP_SCORES;
```

Comment:
The reason I changed this because currently these methods are just being used by collectors as package private method. Having access modifier protected does not have any role in its utilization in the collector. Now as per your earlier comment I moved all the collectors together in search.collector package. Therefore, the methods from HitsThresholdChecker are not accessible from the collectors. That  is why I changed this.